connect/runtime/src/test/java/org/apache/kafka/connect/util/clusters/EmbeddedKafkaCluster.java

```diff
@@ -226,25 +240,37 @@ private void stop(boolean deleteLogDirs, boolean stopZK) {
                 throw new RuntimeException(msg, t);
             }
         }
+    }
 
-        if (deleteLogDirs) {
-            for (KafkaServer broker : brokers) {
-                try {
-                    log.info("Cleaning up kafka log dirs at {}", broker.config().logDirs());
-                    CoreUtils.delete(broker.config().logDirs());
-                } catch (Throwable t) {
-                    String msg = String.format("Could not clean up log dirs for broker at %s",
-                            address(broker));
-                    log.error(msg, t);
-                    throw new RuntimeException(msg, t);
-                }
+    private void awaitBrokerShutdown() {
```

Comment:
Nah, this was just because there was a race in one of our integration tests where we shut down the underlying Kafka cluster and then wait for a fixed time period before asserting that the Connect cluster is unavailable. Making Kafka cluster shutdown synchronous fixed that problem.

(And the method decomposition was just to appease Checkstyle, although I do have to admit I also find it easier to read now.)

Reply:
I gave this a shot, let me know what you think.